packages/@uppy/xhr-upload/src/index.js

```diff
@@ -370,8 +380,13 @@ module.exports = class XHRUpload extends Plugin {
       files.forEach((file, i) => {
         const opts = this.getOptions(file)
 
+        // Set `file.data.type` in the blob to `file.meta.type`,
+        // because we might have detected a more accurate file type in Uppy
+        // https://stackoverflow.com/a/50875615
+        file.data = file.data.slice(0, file.data.size, file.meta.type)
```

Comment:
I would prefer doing `const data = file.data.slice()` here instead of mutating the object, state isn't supposed to be modified outside of setState() calls

Reply:
Of course, forgot about mutation 🤦‍♂ Updated!